chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -555,8 +555,16 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
                     ConsensusData::None | ConsensusData::PoW(_) => match output {
```

Comment:
Instead of branching then joining, please create a `required` for each branch individually.